frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -162,9 +177,40 @@ function VirtualKey:init()
             },
         }
     end
+    if (self.keyboard.shiftmode_keys[self.label] ~= nil  and self.keyboard.shiftmode) or
+    (self.keyboard.umlautmode_keys[self.label] ~= nil and self.keyboard.umlautmode) then
+        self[1].background = Blitbuffer.COLOR_LIGHT_GRAY
+    end
     self.flash_keyboard = G_reader_settings:readSetting("flash_keyboard") ~= false
 end
 
+function VirtualKey:genkeyboardLayoutPopup(parent)
```

Comment:
Seems more logical as a method on the popup instead?
```suggestion
function VirtualKeyPopup:genkeyboardLayoutPopup(parent)
```

Reply:
changed to `VirtualKey:genkeyboardLayoutKeyChars(parent)`